pallets/author/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright 2019-2020 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Pallet that allows block authors to include their identity in a block via an inherent.
+//! Currently the author does not _prove_ their identity, just states it. So it should not be used,
+//! for things like equivocation slashing that require authenticated authorship information.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{decl_error, decl_event, decl_module, decl_storage, ensure, weights::Weight};
+use frame_system::{ensure_none, Config as System};
+use pallet_authorship::EventHandler;
+use parity_scale_codec::{Decode, Encode};
+#[cfg(feature = "std")]
+use sp_inherents::ProvideInherentData;
+use sp_inherents::{InherentData, InherentIdentifier, IsFatalError, ProvideInherent};
+use sp_runtime::{ConsensusEngineId, DigestItem, RuntimeString};
+use sp_std::vec::Vec;
+
+pub trait Config: System {
+	/// Event type used by the runtime.
+	type Event: From<Event<Self>> + Into<<Self as System>::Event>;
+
+	/// Other pallets that want to be informed about block authorship.
+	/// We reuse the `pallet_authorship::EventHandler` for easy compatability with existing pallets.
+	type EventHandler: EventHandler<Self::AccountId, Self::BlockNumber>;
+}
+
+decl_event! {
+	pub enum Event<T> where
+		AccountId = <T as System>::AccountId,
+		BlockNumber = <T as System>::BlockNumber,
+	{
+		/// Set Author. Fields are Author, Block height
+		AuthorSet(AccountId, BlockNumber),
+	}
+}
+
+decl_error! {
+	pub enum Error for Module<T: Config> {
+		/// Author already set in block.
+		AuthorAlreadySet,
+	}
+}
+
+decl_storage! {
+	trait Store for Module<T: Config> as Author {
+		/// Author of current block.
+		Author: Option<T::AccountId>;
+
+		/// Did the author get set in this block?
+		DidUpdate: bool;
```

Comment:
If the Author storage item is None, then don't we already know that the author was not set in this block?

Reply:
Yeah, good point.